### PR TITLE
[#239] Fix timesheet email notification

### PR DESCRIPTION
**Pull request by vaibhav8186**

### CODE REVIEW

File: app/models/time_sheet.rb [hunk 1]
```diff
@@ -871,6 +866,28 @@ def self.load_project_with_id(project_id)
...
+        time_sheet_data = {
+          project_name: time_sheet.project.name,
+          date:         time_sheet.date,
+          from_time:    time_sheet.from_time,
+          to_time:      time_sheet.to_time
+        }
+        user_timesheet << time_sheet_data
+      end
+      if user_timesheet.present?
+        TimesheetRemainderMailer.user_timesheet_for_diffrent_project(user, user_timesheet).deliver_now
```
Comment (swapnilchincholkar) [inline app/models/time_sheet.rb]: @vaibhav8186 What will happen if user has 4 entries of timesheet either for 1 project OR for multiple projects which are not assigned to him? 
Make sure only 1 email should be delivered per day with combined details of all such not assigned projects.

File: app/models/time_sheet.rb [hunk 2]
```diff
@@ -871,6 +866,28 @@ def self.load_project_with_id(project_id)
     Project.find_by(id: project_id)
   end
 
+  def self.get_users_and_timesheet_who_have_filled_timesheet_for_diffrent_project
+    User.approved.each do | user |
+      user_timesheet = []
+      time_sheets    = user.time_sheets.where(date: Date.today - 1)
```
Comment (swapnilchincholkar) [inline app/models/time_sheet.rb]: @vaibhav8186 Here if user creates timesheet for 4 days before work, then in cron we will miss it right? Should we query as "created yesterday" irrespective of 'date' assigned to record?

File: app/views/time_sheets/edit_timesheet.html.haml [hunk 3]
```diff
@@ -12,7 +12,7 @@
     %tbody
       = f.fields_for :time_sheets, @time_sheets do |time_sheet|
         %tr
-          %td= time_sheet.select :project_id, options_for_select(@user.projects.pluck(:name, :id), time_sheet.object.project_id), {}, {style: "margin-bottom: 16px"}
+          %td= time_sheet.select :project_id, options_for_select(active_project_ids(@user), selected: time_sheet.object.project_id), {}, {style: "margin-bottom: 16px"}
```
Comment (swapnilchincholkar) [inline app/views/time_sheets/edit_timesheet.html.haml]: @vaibhav8186 Make sure projects are sorted by name. i.e projects should be listed in 2 groups (assigned-sorted by name, not assigned-sorted by name).

File: app/models/time_sheet.rb (hunk 1 above)
Comment (vaibhav8186) [reply app/models/time_sheet.rb]: yes. Combined details will go in only one email

File: app/models/time_sheet.rb (hunk 2 above)
Comment (vaibhav8186) [reply app/models/time_sheet.rb]: yes. We have to change it on created_at attribute

File: app/views/time_sheets/edit_timesheet.html.haml (hunk 3 above)
Comment (vaibhav8186) [reply app/views/time_sheets/edit_timesheet.html.haml]: yes..for that I have added scope -all_active, previously it was not there.

**Pull request merged**